### PR TITLE
FFS-4003: Refactor logic for creating education Activity Hub cards and progress calculation

### PR DESCRIPTION
**Pull request by daphnegold**
### Phase 1: Edge case fixes

- Activity Hub education cards work for summer carryover in long reporting windows
- Fixed the spring & fall with no-summer sandwich case
- Added Morgan Testuser to `/demo` to test the no-summer sandwich

### Phase 2: Logic consolidation

- New `EducationReportingMonthResolver` to centralize education reporting-month logic as that logic being split across cards, progress, routing, etc contributed to the P1 bugs

## Context for reviewers
I didn't include the suggested [FFS-3955](https://jiraent.cms.gov/browse/FFS-3955) Activity Hub ordering change in this PR or any grander refactors outside education. It felt too "kitchen sink" as it affects all card types. [FFS-3955](https://jiraent.cms.gov/browse/FFS-3955) is probably the right place to do any cleanup if we'd like.

## Acceptance testing
Tag product and design in Slack for acceptance: @emmy-acceptance-testers

- [ ] No acceptance testing needed
  * This change will not affect the user experience (bugfix, dependency updates, etc.)
- [x] Acceptance testing prior to merge
  * This change can be verified visually via screenshots attached below or by sending a link to a local development environment to the acceptance tester
  * Acceptance testing should be done by **design** for visual changes, **product** for behavior/logic changes, **or both** for changes that impact both.
- [ ] Acceptance testing in PR Environment
  * This change can be verified in a PR environment. Run [this Github Action](https://github.com/DSACMS/iv-cbv-payroll/actions/workflows/ci-app-pr-environment-checks.yml) with the existing PR and most recent git sha.
- [ ] Acceptance testing after merge
  * This change is hard to test locally, so we'll test it in the demo environment (deployed automatically after merge.)
  * Make sure to notify the team once this PR is merged so we don't inadvertently deploy the unaccepted change to production. (e.g. `:alert: Deploy block! @ffs-eng I just merged PR [#123] and will be doing acceptance testing in demo - please don't deploy until I'm finished!`)

## Preview environment
- Link to demo: https://p-1708.navapbc.cloud/demo
- Deployed commit: cccd047b5060afae133613f63b96cdb035e3c7c8

### CODE REVIEW

**Comment by rutvigupta-design** [comment]
Looks great to me, thanks for adding the new test case - made it super easy to test!

File: app/app/services/education_reporting_month_resolver.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+class EducationReportingMonthResolver
```
Comment (tdooner) [inline app/app/services/education_reporting_month_resolver.rb:1]: I know naming things is hard but a comment about what a "ResolvedMonth" would be might make this a bit more clear. I wasn't sure why the months had to be "resolved"... I assume that is the process of converting a list of terms, combined with a reporting period, into a list of months that are either verified or need verification.

File: app/spec/services/activity_flow_progress_calculator_spec.rb [hunk 2]
```diff
@@ -1101,6 +1101,41 @@ def create_summary(month:, **attrs)
...
+          term_end: Date.new(2025, 12, 15)
+        )
+      end
+
+      it "meets routing requirements with spring carryover and fall enrollment" do
+        expect(progress.meets_requirements).to be(true)
+        expect(progress.meets_routing_requirements).to be(true)
+      end
+    end
+
```
Comment (tdooner) [inline app/spec/services/activity_flow_progress_calculator_spec.rb:1138]: Ty for fixing this.

**Review by tdooner — APPROVED** [review]

File: app/app/services/education_reporting_month_resolver.rb (hunk 1 above)
Comment (daphnegold) [reply app/app/services/education_reporting_month_resolver.rb:1]: I agree, I went back and forth on it myself. I think I found a better name and updated for consistency, but left a comment even so.
